x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/rules/action/TransportPutQueryRulesetAction.java

```diff
@@ -11,24 +11,51 @@
 import org.elasticsearch.action.support.ActionFilters;
 import org.elasticsearch.action.support.HandledTransportAction;
 import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.cluster.service.ClusterService;
 import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.settings.ClusterSettings;
+import org.elasticsearch.common.settings.Setting;
 import org.elasticsearch.tasks.Task;
 import org.elasticsearch.transport.TransportService;
+import org.elasticsearch.xpack.application.rules.QueryRulesConfig;
 import org.elasticsearch.xpack.application.rules.QueryRulesIndexService;
 import org.elasticsearch.xpack.application.rules.QueryRuleset;
 
+import java.util.Objects;
+
 public class TransportPutQueryRulesetAction extends HandledTransportAction<PutQueryRulesetAction.Request, PutQueryRulesetAction.Response> {
     protected final QueryRulesIndexService systemIndexService;
+    private final ClusterSettings clusterSettings;
 
     @Inject
-    public TransportPutQueryRulesetAction(TransportService transportService, ActionFilters actionFilters, Client client) {
+    public TransportPutQueryRulesetAction(
+        TransportService transportService,
+        ClusterService clusterService,
+        ActionFilters actionFilters,
+        Client client
+    ) {
         super(PutQueryRulesetAction.NAME, transportService, actionFilters, PutQueryRulesetAction.Request::new);
         this.systemIndexService = new QueryRulesIndexService(client);
+        this.clusterSettings = clusterService.getClusterSettings();
     }
 
     @Override
     protected void doExecute(Task task, PutQueryRulesetAction.Request request, ActionListener<PutQueryRulesetAction.Response> listener) {
         QueryRuleset queryRuleset = request.queryRuleset();
+        @SuppressWarnings("unchecked")
```

Comment:
Probably it would be better to move this check to the `QueryRulesIndexService`. Other operations that use it would benefit from it as well, and the Transport action wouldn't have to deal with this.

Reply:
Good suggestion, done ✔️